src/shared/src/types.rs

```diff
@@ -100,3 +100,57 @@ pub mod custom_token {
         Icrc(LedgerId),
     }
 }
+
+/// Types specifics to the user profile.
+pub mod user_profile {
+    use candid::{CandidType, Deserialize, Principal};
+    use std::collections::BTreeMap;
+
+    pub type CredentialType = String;
+
+    #[derive(CandidType, Deserialize, Clone, Eq, PartialEq, Debug)]
+    pub struct UserCredential {
+        pub credential_type: CredentialType,
+        pub verified_date_timestamp: Option<u64>,
+        pub expire_date_timestamp: Option<u64>,
+    }
+
+    // Used in the endpoint
+    #[derive(CandidType, Deserialize, Clone, Eq, PartialEq, Debug)]
+    pub struct UserProfile {
+        pub credentials: Vec<UserCredential>,
+        pub created_timestamp: u64,
+        pub updated_timestamp: u64,
+    }
+
+    #[derive(CandidType, Deserialize, Clone, Eq, PartialEq, Debug)]
+    pub struct UserProfileStorage {
```

Comment:
What's that struct for?

Reply:
This is the type stored in stable memory. `UserProfile` is the one returned to the client.

I had to use two because I couldn't use a Map of credential_type to CredentialSummary in Candid and instead return a vector of CredentialSummary.

Reply:
Ah got it! For consistency, should we call it `StoredUserProfile`? It's a bit different but, we already use  `StoredPrincipal` to store principal as key of the maps.

```
type CustomTokenMap = StableBTreeMap<StoredPrincipal, Candid<Vec<CustomToken>>, VMem>;

MEMORY_MANAGER.with(|mm| State {
...
            custom_token: CustomTokenMap::init(mm.borrow().get(USER_CUSTOM_TOKEN_MEMORY_ID)),
        })
```

Reply:
Ah, sure, that makes sense.

I'll rename it in another PR.